crates/edr_provider/src/data/call.rs

```diff
@@ -64,21 +68,23 @@ pub(super) fn run_call(args: RunCallArgs<'_>) -> Result<ResultAndState, Provider
 }
 
 /// Execute a transaction as a call. Returns the gas used and the output.
-pub(super) fn run_call_and_handle_errors(
+pub(super) fn run_call_and_handle_errors<LoggerErrorT: Debug>(
     run_call_args: RunCallArgs<'_>,
     transaction_hash: &B256,
-) -> Result<(u64, Bytes), ProviderError> {
+) -> Result<Result<(u64, Bytes), TransactionFailure>, ProviderError<LoggerErrorT>> {
```

Comment:
It seems like my comment got lost here, if you could flatten the `Result`, that'd be great, but not a blocker.

Reply:
See: https://github.com/NomicFoundation/hardhat/pull/4781#discussion_r1465900562

Reply:
Addressed in https://github.com/NomicFoundation/hardhat/pull/4781/commits/c5a95d041b47d6e88ba5f5e0ba471ddb1d827df4